soroban-sdk/src/address.rs

```diff
@@ -226,49 +223,55 @@ impl Address {
         self.env.require_auth(self);
     }
 
-    /// Creates an `Address` corresponding to the provided contract identifier.
+    /// Creates an `Address` corresponding to the provided Stellar strkey.
     ///
-    /// Prefer using the `Address` directly as input or output argument. Only
-    /// use this in special cases, for example to get an Address of a freshly
-    /// deployed contract.
+    /// The only supported strkey types are account keys (`G...`) and contract keys (`C...`). Any
+    /// other valid or invalid strkey will cause this to panic.
     ///
-    /// TODO: Replace this function in its pub form with a function that accepts
-    /// a strkey instead. Dependent on <https://github.com/stellar/rs-stellar-strkey/issues/56>.
-    pub fn from_contract_id(contract_id: &BytesN<32>) -> Self {
-        let env = contract_id.env();
+    /// Prefer using the `Address` directly as input or output argument. Only
+    /// use this in special cases when addresses need to be shared between
+    /// different environments (e.g. different chains).
+    pub fn from_strkey(strkey: &String) -> Self {
```

Comment:
Imo we should implement the FromStr trait that is in the core lib, which will result in a `from_str` function. Also, this function should be fallible? Or does the host function panic?
```suggestion
    pub fn from_str(str: &String) -> Self {
```

Reply:
Oh, I guess we can't implement FromStr because of the need for the accompanying env.

Reply:
Host function causes panic, as per usual for the host functions (besides `try_call`).

I'm not sure about implementing the trait - strkey is not an immediately obvious string format and I would prefer to keep it explicit. Things get especially confusing for `from_bytes` because one could expect some binary serialization format (e.g. XDR) and mis-use this for e.g. hashing.

Reply:
```suggestion
    pub fn from_string(strkey: &String) -> Self {
```
?

Reply:
> Host function causes panic, as per usual for the host functions

Host functions are supposed to only panic if there's a way to avoid the panic in the application. There's no way to avoid the panic here.

Reply:
I'm not opposed to e.g. returning `Void`, but in general we haven't followed this too closely (e.g. crypto functions panic). I can come back to this in another iteration - there is a bunch of changes we need to do besides this one, so I don't want to be blocked on another host change.

Reply:
> I can come back to this in another iteration 

👍🏻

Reply:
Sounds good, I did the renames, but not the traits (because we pass in SDK types).